images/win/scripts/ImageHelpers/InstallHelpers.ps1

```diff
@@ -161,50 +161,63 @@ Hashtable for service arguments
     }
 }
 
-function Install-VsixExtension
+function Start-DownloadWithRetry
 {
-    Param
-    (
+    param (
+        [Parameter(Mandatory)]
         [string] $Url,
-        [Parameter(Mandatory = $true)]
+        [string] $DownloadPath = "${env:Temp}",
+        [Parameter(Mandatory)]
         [string] $Name,
-        [string] $FilePath,
-        [Parameter(Mandatory = $true)]
-        [string] $VSversion,
-        [int] $retries = 20,
-        [switch] $InstallOnly
-    )
-
-    if (!$InstallOnly)
-    {
-        $FilePath = "${env:Temp}\$Name"
-
-        while($retries -gt 0)
+        [int] $retries = 20
+        )
+    $FilePath = Join-Path $DownloadPath $Name
+    #Default retry logic for the package.
+    while($retries -gt 0)
         {
             try
             {
-                Write-Host "Downloading $Name..."
+                Write-Host "Downloading package from: $Url to path $FilePath ."
                 (New-Object System.Net.WebClient).DownloadFile($Url, $FilePath)
                 break
             }
             catch
             {
-                Write-Host "There is an error during $Name downloading"
+                Write-Host "There is an error during package downloading"
                 $_
-
                 $retries--
 
                 if ($retries -eq 0)
                 {
-                    Write-Host "File can't be downloaded"
+                    Write-Host "File can't be downloaded. Please try later or check that file exists by url: $Url"
                     $_
                     exit 1
                 }
-
                 Write-Host "Waiting 30 seconds before retrying. Retries left: $retries"
                 Start-Sleep -Seconds 30
             }
         }
+   return $FilePath
+}
+
+
+function Install-VsixExtension
+{
+    Param
+    (
+        [string] $Url,
+        [Parameter(Mandatory = $true)]
+        [string] $Name,
+        [string] $FilePath,
+        [Parameter(Mandatory = $true)]
```

Comment:
nitpick: mandatory args are usually in the beginning so it's easier to read the code.